easy-ui-react/src/TextField/TextField.module.scss

```diff
@@ -0,0 +1,274 @@
+    color: component-token("textfield", "color.text.engaged");
+    border: design-token("shape.border_width.1") solid
+      component-token("textfield", "color.border.engaged");
+  }
+
+  &:focus {
+    box-shadow: component-token("textfield", "box_shadow");
+  }
+
+  &::placeholder {
+    color: component-token("textfield", "color.text.gray.resting");
+  }
+
+  &:disabled {
+    background-color: component-token("textfield", "color.background.disabled");
+    color: component-token("textfield", "color.text.disabled");
+  }
+}
+
+.errorInput {
+  color: component-token("textfield", "color.text.danger");
+  border-color: component-token("textfield", "color.border.danger");
+
+  &:focus,
+  &:active {
+    border-color: component-token("textfield", "color.border.danger");
+  }
+}
+
+.passwordInput {
+  border-radius: component-token("textfield", "border_radius") 0 0
+    component-token("textfield", "border_radius");
+  border-right: none;
+  padding-right: design-token("space.0.5");
+  &:focus,
+  &:active {
+    padding-right: calc(
+      #{design-token("space.0.5")} - #{design-token("shape.border_width.1")}
+    );
+    border-radius: component-token("textfield", "border_radius");
+  }
+}
+
+.inputSizeSm {
+  @include font-style("body2");
+  padding: calc(
+      #{design-token("space.1")} - #{design-token("shape.border_width.1")}
+    )
+    calc(#{design-token("space.1.5")} - #{design-token("shape.border_width.1")});
+  max-height: component-token("textfield", "max-height-sm");
+}
+
+.inputSizeLg {
+  padding: calc(
+    #{design-token("space.2")} - #{design-token("shape.border_width.1")}
+  );
+}
+
+.passwordBtn {
+  background-color: component-token("textfield", "color.background");
+  color: component-token("textfield", "color.text.gray.resting");
+  border-radius: 0 component-token("textfield", "border_radius")
+    component-token("textfield", "border_radius") 0;
+  border: design-token("shape.border_width.1") solid
+    component-token("textfield", "color.border.resting");
+  border-left: none;
+  margin: 0;
+  padding: calc(
+    #{design-token("space.1.5")} - #{design-token("shape.border_width.1")}
+  );
+  padding-left: design-token("space.1.5");
+
+  &:focus,
+  &:active {
+    border: design-token("shape.border_width.1") solid
+      component-token("textfield", "color.border.engaged");
+    padding-left: calc(
+      #{design-token("space.1.5")} - #{design-token("shape.border_width.1")}
+    );
+    border-radius: component-token("textfield", "border_radius");
+  }
+
+  &:focus {
+    box-shadow: component-token("textfield", "box_shadow");
+    outline: none;
+    z-index: component-token("textfield", "z-index");
+  }
+
+  &:disabled {
+    background-color: component-token("textfield", "color.background.disabled");
+  }
+}
+
+.passwordBtnError {
+  border-color: component-token("textfield", "color.border.danger");
+  &:focus,
+  &:active {
+    border-color: component-token("textfield", "color.border.danger");
+  }
+}
```

Comment:
**suggestion:** you could reduce the border-color declarations by combining selectors
```suggestion
.passwordBtnError,
.passwordBtnError:focus,
.passwordBtnError:active {
  border-color: component-token("textfield", "color.border.danger");
}
```

Reply:
good call